examples/go/Dockerfile

```diff
@@ -1,12 +1,28 @@
-FROM golang:1.12 as build-env
+# our builder image
+FROM golang:1.13.1-alpine AS builder
 
+# default app directory
 WORKDIR /go/src/app
+
+# copy-in our module info
+COPY go.mod go.sum /go/src/app/
+RUN go mod download
+
+# add our binary sources
 ADD . /go/src/app
 
-RUN go get -d -v ./...
+# build our binary
+RUN CGO_ENABLED=0 go build -a -installsuffix cgo -o /go/bin/app
 
-RUN go build -o /go/bin/app
+# ---
 
 FROM gcr.io/distroless/base
-COPY --from=build-env /go/bin/app /
-CMD ["/app"]
+
+# copy our built binary
+COPY --from=builder --chown=nonroot /go/bin/app /app
```

Comment:
`COPY` should be the last command to better use Docker's caching. `USER` and `CMD` never change so they should appear before.

Reply:
Fixed.